README.md

```diff
@@ -2,7 +2,157 @@
 
 # Prelude
 
+ASON is a data oriented algorithm designed for compact and speedy storage of AssemblyScript objects in a binary format.
 
+There are many Serialization methods out there that can turn a conceptual object into a buffer of some kind, like JSON and protobuf. Each one of these methods has their benefits, and none of them can take advantage of runtime type information provided by the AssemblyScript compiler and the runtime type information.
+
+Another problem with serialization methods is that they tend to be "tree-like" instead of data oriented. To make matters worse, there are no ways for developers to determine field definitions on a given generic reference object at runtime. This is where the ASON algorithm comes in handy.
+
+Instead of storing a tree structure, ASON stores a set of instructions to assemble an object from scratch using the smallest amount of space possible.
+
+# How To Use
+
+Install from npm:
+
+```
+npm install @ason/assembly
+```
+
+Modify your asconfig to include the transform:
+
+```ts
+{
+  ...
+  "transform": ["@ason/transform"]
+}
+```
+
+Import the library and serialize away!
+
+```ts
+import { ASON } from "@ason/assembly";
+
+// serialize can determine type information
+let buffer = ASON.serialize([29, 6, 4] as StaticArray<u8>);
```

Comment:
I'd suggest adding the type for buffer here, I know it is inferred, but it helps me understand the ASON API if you add it